### ISSUE TITLE
Planet example doesn't work

### ISSUE THREAD

**Issue by asterite** [joeyrobert/drawille-cr#1]
This project looks really cool!

However, the `planet` example doesn't work for me...

```
$ crystal examples/planet.cr
Unhandled exception: Index out of bounds (IndexError)
  from /usr/local/Cellar/crystal/0.33.0/src/indexable.cr:73:20 in '[]'
  from lib/stumpy_core/src/stumpy_core/canvas.cr:62:7 in 'get'
  from lib/stumpy_core/src/stumpy_core/canvas.cr:73:7 in '[]'
  from examples/planet.cr:47:17 in 'run_animation'
  from examples/planet.cr:83:1 in '__crystal_main'
  from /usr/local/Cellar/crystal/0.33.0/src/crystal/main.cr:106:5 in 'main_user_code'
  from /usr/local/Cellar/crystal/0.33.0/src/crystal/main.cr:92:7 in 'main'
  from /usr/local/Cellar/crystal/0.33.0/src/crystal/main.cr:115:3 in 'main'
```

**Comment by joeyrobert** [comment]
@asterite Thanks for checking out this project and reporting the bug. Obviously I'm a big fan of Crystal.

About this issue, I noticed the planet renderer was failing in specific terminal sizes and thought I had fixed it, but wasn't very thorough in my testing. I was struggling with the math about why this approach was leading to NaNs for indexes, but it seems catching the `IndexError` and moving on still adequately renders the planet. I added some smoketests to validate this.

Try `crystal examples/planet_cli.cr`, it should work for you now.
